redaxo/src/core/lib/login/api_has_user_session.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+/**
+ * @package redaxo\core
+ *
+ * @internal
+ */
+class rex_api_has_user_session extends rex_api_function
+{
+    public function execute()
+    {
+        if (!rex_request::isHttps()) {
+            throw new rex_api_exception(sprintf('https is required'));
+        }
+
+        $user = rex::getUser();
+        if (!$user) {
+            rex_response::sendContent(json_encode(false), 'application/json');
+            exit();
+        }
+
+        $perm = rex_get('perm');
+        if ($perm) {
+            rex_response::sendContent(json_encode($user->hasPerm($perm)), 'application/json');
+            exit();
+        }
+
+        rex_response::sendContent(json_encode(true), 'application/json');
```

Comment:
Man könnte überlegen, mal einen Shortcut in rex_response zu erstellen, um eine JSON-Antwort zu senden.

Reply:
https://github.com/redaxo/redaxo/issues/2118